### PR TITLE
Fixes invalid paths when using rsync with Windows

### PR DESCRIPTION
**Pull request by cdrage**
This PR fixes the invalid paths that are being passed into OpenShift due
to using `FromSlash` that defaults to whatever backslash / forwardslash
the OS uses.

This changes it so that `ToSlash` is used for transfering as *all*
containers in OpenShift are Linux-based and thus forward-slashes are
required.

Also added:
 - Checks to see if a directory is empty and error out.

To test this PR out:
 - Have a Windows-based OS
 - `git clone https://github.com/openshift/nodejs-ex`
 - `odo create nodejs --context ~/nodejs-ex`
 - `odo push -v 4 --context ~/nodejs-ex`
 - `odo watch -v 4 --context ~/nodejs-ex`

Closes issue https://github.com/openshift/odo/issues/1429

### CODE REVIEW

**Comment by cdrage** [comment]
Please ignore the `cdrage:ignore-behaviour-windows` name of the branch. It's related to this fix :)  This fixes #1429 not #1630

File: pkg/component/component.go [hunk 1]
```diff
@@ -626,6 +626,15 @@ func ApplyConfigCreateUrl(client *occlient.Client, componentConfig config.LocalC
 //	Error if any
 func PushLocal(client *occlient.Client, componentName string, applicationName string, path string, out io.Writer, files []string, delFiles []string, isForcePush bool, globExps []string, show bool) error {
 	glog.V(4).Infof("PushLocal: componentName: %s, applicationName: %s, path: %s, files: %s, delFiles: %s, isForcePush: %+v", componentName, applicationName, path, files, delFiles, isForcePush)
+
+	// Edge case: check to see that the path is NOT empty.
```
Comment (mohammedzee1000) [inline pkg/component/component.go]: +1

**Comment by girishramnani** [comment]
if possible a unit test for `CopyFile` with different slashes would be great here @cdrage

**Comment by cdrage** [comment]
@girishramnani I wrote a brief test for CopyFile but found it too cumbersome to actually emulate the copying over the files.

This is covered via integration tests. But I'll look into other ways to test this functionality!

**Comment by girishramnani** [comment]
```
Client Version: version.Info{Major:"4", Minor:"0+", GitVersion:"v4.0.0-alpha.0+b93a8e6-2141-dirty", GitCommit:"b93a8e6", GitTreeState:"", BuildDate:"2019-04-23T21:16:33Z", GoVersion:"", Compiler:"", Platform:""}
error: You must be logged in to the server (the server has asked for the client to provide credentials)
make[1]: *** [configure-installer-tests-cluster] Error 1
```
Login failure
/retest
